lib/jekyll-readme-index/generator.rb

```diff
@@ -67,7 +68,12 @@ def dir_has_index?(relative_path)
 
     # Regexp to match a file path against to detect if the given file is a README
     def readme_regex
-      @readme_regex ||= %r!/readme(#{Regexp.union(markdown_converter.extname_list)})$!i
+      regex = option(REGEX_KEY)
+      if regex
+        @readme_regex ||= Regexp.new(regex)
```

Comment:
Can I pass flags to the `Regex` constructor so that I can make the user-provided regex case-insensitive by default as well?